### PR TITLE
added versioning to TempusController & Stats; changed version type to string

### PR DESCRIPTION
**Pull request by yuval-weiss**

### CODE REVIEW

**Comment by RedFox20** [comment]
Looks good, I think you need small update to TempusPool.ts as well for completeness sake

File: contracts/TempusController.sol [hunk 1]
```diff
@@ -21,6 +21,8 @@ contract TempusController is ReentrancyGuard, Ownable {
     using UntrustedERC20 for IERC20;
     using AMMBalancesHelper for uint256[];
 
+    string public constant version = "1.0.0";
```
Comment (axic) [inline contracts/TempusController.sol:24]: String is not a very good choice for version numbers, should use semver. Semver defines what exact number means, ie. on major=1 any change to minor is fine and non breaking, however programatically finding this out means a lot of text parsing, which is not ideal off-chain (but doable) and highly inefficient on-chain.

Comment (axic) [reply contracts/TempusController.sol:24]: A better option is the usual way to combine these into a single number: `(major << 16) | (minor << 8) | patch`.

File: contracts/stats/Stats.sol [hunk 2]
```diff
@@ -15,6 +15,8 @@ contract Stats is ITokenPairPriceFeed, ChainlinkTokenPairPriceFeed {
     using Fixed256xVar for uint256;
     using AMMBalancesHelper for uint256[];
 
+    string public constant version = "1.0.0";
```
Comment (RedFox20) [inline contracts/stats/Stats.sol:18]: What about Stats contract?

File: test/TempusPool.Deploy.test.ts [hunk 3]
```diff
@@ -16,7 +16,7 @@ describeForEachPool("TempusPool Deploy", (testPool:PoolTestFixture) =>
 
   it("Version is correct", async () =>
   {
-    expect(await pool.version()).to.equal(1);
+    expect(await pool.version()).to.equal("1.0.0");
```
Comment (RedFox20) [inline test/TempusPool.Deploy.test.ts:19]: I guess this needs an update now?

Comment (yuval-weiss) [reply test/TempusPool.Deploy.test.ts]: yup, done.

File: test/TempusController.test.ts [hunk 4]
```diff
@@ -50,6 +50,21 @@ describeForEachPool("TempusController", (testPool:PoolTestFixture) =>
     expect(+await pool.yieldShare.balanceOf(controller.address)).to.be.lessThan(2e-18, "No funds should remain in controller");
   }
 
+  describe("deploy", async () => {
+    it("Version is correct", async () =>
+  {
```
Comment (RedFox20) [inline test/TempusController.test.ts:55]: Some whitespace issues here, but otherwise looks great

File: contracts/stats/Stats.sol (hunk 2 above)
Comment (yuval-weiss) [reply contracts/stats/Stats.sol]: What about it? I added versioning to Stats as well..

File: test/stats/Stats.test.ts [hunk 5]
```diff
@@ -32,6 +32,14 @@ describeForEachPool("Stats", (testPool:PoolTestFixture) =>
     stats = await Stats.create();
   });
 
+  it("Version is correct", async () =>
+  {
+    const { major, minor, patch } = await stats.version();
```
Comment (RedFox20) [inline test/stats/Stats.test.ts:37]: That's what's kind of annoying here, would be easy to just check for "1.0.0" instead

**Review by RedFox20 — APPROVED** [review]

**Pull request merged**